The provider crashes

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and docker Provider) Version

```
Terraform v1.3.3
on linux_amd64
+ provider registry.terraform.io/kreuzwerker/docker v2.23.0
```

Docker version:

```
Server: Docker Engine - Community
 Engine:
  Version:          20.10.21
```


### Affected Resource(s)

* `docker_container`

### Terraform Configuration Files

```hcl
resource "docker_container" "provisioner_container" {
  name = "terraform-eks-provisioner-${substr(sha1(timestamp()), 0, 4)}"
  image = docker_image.provisioner_image.image_id
  mounts {
    source = local_file.kubeconfig.filename
    target = "/etc/kubectl.yaml"
    type = "bind"
    read_only = true
  }
  env = [
    "KUBECONFIG=/etc/kubectl.yaml"
  ]
  # the docker provider does not wait for containers that are supposed to exit
  # even if another resource uses its exit_code attribute (it will be just
  # null) so we start a container that keeps running for a short time after it
  # finished its task
  must_run = true
  wait = true
  wait_timeout = 60
  healthcheck {
    test = ["CMD", "bash", "-c", "[ -f /workspace/DONE ]"]
    interval = "1s"
  }
}
```

### Debug Output

The only vaguely relevant part ,I don't think it has anything to do with the panic.

```
module.eks_cluster_calico.docker_container.provisioner_container: Still creating... [1m0s elapsed]
2022-11-17T14:09:38.077Z [INFO]  provider.terraform-provider-docker_v2.23.0: 2022/11/17 14:09:38 [DEBUG] waiting for container healthy state: timestamp=2022-11-17T14:09:38.077Z
2022-11-17T14:09:38.538Z [INFO]  provider.terraform-provider-docker_v2.23.0: 2022/11/17 14:09:38 [ERROR] Container 948d0180a71e34429661b8afe9f40d861c077dbee2db4ef5e2d0e72d3526d94b failed to be in healthy state in time: timestamp=2022-11-17T14:09:38.537Z
2022-11-17T14:09:38.541Z [ERROR] provider.terraform-provider-docker_v2.23.0: Response contains error diagnostic: @caller=github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/diag/diagnostics.go:55 @module=sdk.proto diagnostic_severity=ERROR tf_proto_version=5.3 tf_resource_type=docker_container diagnostic_detail= diagnostic_summary="container failed to be in healthy state" tf_provider_addr=provider tf_req_id=24f4f9e8-b941-9dd4-3bb2-33fcc4b9782e tf_rpc=ApplyResourceChange timestamp=2022-11-17T14:09:38.540Z
2022-11-17T14:09:38.549Z [ERROR] vertex "module.eks_cluster_calico.docker_container.provisioner_container" error: container failed to be in healthy state
```

Afterwards it's just terraform finishing the run and then the status report with the provider crash logs is printed.

### Panic Output

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0xc45f2f]

goroutine 48 [running]:
github.com/terraform-providers/terraform-provider-docker/internal/provider.resourceDockerContainerCreate.func1(0x0?)
	github.com/terraform-providers/terraform-provider-docker/internal/provider/resource_docker_container_funcs.go:518 +0x1af
created by github.com/terraform-providers/terraform-provider-docker/internal/provider.resourceDockerContainerCreate
	github.com/terraform-providers/terraform-provider-docker/internal/provider/resource_docker_container_funcs.go:531 +0x2c6e
Error: The terraform-provider-docker_v2.23.0 plugin crashed!
```

### Expected Behaviour

A little bit of context: This container is part of a module that spins up an EKS cluster. This container is supposed to configure a few things on the cluster before the node groups are added. The container supposed to be short lived, it runs kubectl and Helm. This docker-based solution is an attempt at not having to shell out using a local-exec provisioner.

However, I couldn't find a combination of resource attributes (must_run, wait, etc.) that would make the resource wait for the container to exit. The workaround is launching a container that will be kept running with sleep for a few minutes so the container resource can wait for it to become healthy (when it finished its task). The error occurs after the resource timed out waiting because of `wait_timeout = 60`.

```
module.eks_cluster_calico.docker_container.provisioner_container: Creating...
module.eks_cluster_calico.docker_container.provisioner_container: Still creating... [10s elapsed]
module.eks_cluster_calico.docker_container.provisioner_container: Still creating... [20s elapsed]
module.eks_cluster_calico.docker_container.provisioner_container: Still creating... [30s elapsed]
module.eks_cluster_calico.docker_container.provisioner_container: Still creating... [40s elapsed]
module.eks_cluster_calico.docker_container.provisioner_container: Still creating... [50s elapsed]
module.eks_cluster_calico.docker_container.provisioner_container: Still creating... [1m0s elapsed]

Error: container failed to be in healthy state
│ 
│   with module.eks_cluster_calico.docker_container.provisioner_container,
│   on ../../../modules/eks_cluster/provisioner.tf line 65, in resource "docker_container" "provisioner_container":
│   65: resource "docker_container" "provisioner_container" {
│ 
╵

Stack trace from the terraform-provider-docker_v2.23.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0xc45f2f]

snip
```

### Actual Behaviour

<!--- What actually happened? --->

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your setup that we should know? For example: Running the docker daemon with TLS? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Such as vendor documentation?
--->

* #0000

--- a/internal/provider/resource_docker_container_funcs.go
+++ b/internal/provider/resource_docker_container_funcs.go
@@ -384,7 +384,7 @@ func resourceDockerContainerCreate(ctx context.Context, d *schema.ResourceData, 
 	if retContainer, err = client.ContainerCreate(ctx, config, hostConfig, networkingConfig, nil, d.Get("name").(string)); err != nil {
 		return diag.Errorf("Unable to create container: %s", err)
 	}
-
+	log.Printf("[INFO] retContainer %#v", retContainer)
 	d.SetId(retContainer.ID)
 
 	// But overwrite them with the future ones, if set
@@ -504,7 +504,8 @@ func resourceDockerContainerCreate(ctx context.Context, d *schema.ResourceData, 
 					if err != nil {
 						result <- fmt.Errorf("error inspecting container state: %s", err)
 					}
-					if infos.State.Health.Status == types.Healthy {
+					//infos.ContainerJSONBase.State.Health is only set when there is a healthcheck defined on the container resource
+					if infos.ContainerJSONBase.State.Health.Status == types.Healthy {
 						log.Printf("[DEBUG] container state is healthy")
 						break
 					}

